.github/workflows/run-unit-tests.yml

```diff
@@ -358,3 +374,30 @@ jobs:
         with:
           swift-version: ${{ matrix.swift-version }}
           copy-files: ${{ inputs.android_copy_files }}
+          swift-test-flags: ${{ env.ENABLE_ALL_TRAITS }}
+
+  disabled-traits-unit:
+    if: ${{ !(github.event.pull_request.draft || false) && inputs.with_disabled_traits }}
+    runs-on: ubuntu-latest
+    container: swift:6.2.0-noble
+    timeout-minutes: 30
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v6
+      # NOTE: Portions of this workflow are adapted from
+      # https://github.com/apple/swift-nio/pull/3159/
+      - name: Install linux dependencies
+        run: apt-get update -y -q && apt-get install -y -q curl && apt-get install -y -q jq
+      - name: Install SDK
+        run: |
+          version="$(swiftc --version | head -n1)"
+          tag="$(curl -sL "https://raw.githubusercontent.com/swiftwasm/swift-sdk-index/refs/heads/main/v1/tag-by-version.json" | jq -e -r --arg v "$version" '.[$v] | .[-1]')"
+          curl -sL "https://raw.githubusercontent.com/swiftwasm/swift-sdk-index/refs/heads/main/v1/builds/$tag.json" | jq -r '.["swift-sdks"]["wasm32-unknown-wasip1-threads"] | "swift sdk install \"\(.url)\" --checksum \"\(.checksum)\""' | sh -x
+      - name: Check Swift compatibility
+        id: swift-check
+        uses: vapor/ci/.github/actions/check-compatible-swift@main
+        with:
+          package_root: ${{ inputs.package_root }}
+      - name: Build
+        if: ${{ steps.swift-check.outputs.swift-compatible == 'true' }}
+        run: swift build --swift-sdk wasm32-unknown-wasip1-threads ${PACKAGE_PATH} --disable-default-traits ${EXTRA_FLAGS} ${WARNINGS_AS_ERRORS}
```

Comment:
I think we'll need an extra flags to customise which tests to run

Reply:
And why the WASM SDK?

Reply:
I'm not sure what flags you mean? If you want to avoid running tests inside of the traits those won't even be compiled if behind the trait

And WASM just because
> What we should actually have is a job like WASM that runs the tests with --disable-default-traits that we can pass some flags into